File: tests/functional/Bots/Python/Skills/CodeFirst/WaterfallSkillBot/dialogs/cards/card_options.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from enum import Enum


class CardOptions(str, Enum):
    ADAPTIVE_CARD_BOT_ACTION = "AdaptiveCardBotAction"
    ADAPTIVE_CARD_TEAMS_TASK_MODULE = "AdaptiveCardTeamsTaskModule"
    ADAPTIVE_CARD_SUBMIT_ACTION = "AdaptiveCardSubmitAction"
    HERO = "Hero"
    THUMBNAIL = "Thumbnail"
    RECEIPT = "Receipt"
    SIGN_IN = "Signin"
    CAROUSEL = "Carousel"
    LIST = "List"
    O365 = "O365"
    TEAMS_FILE_CONSENT = "TeamsFileConsent"
    ANIMATION = "Animation"
    AUDIO = "Audio"
    VIDEO = "Video"
    ADAPTIVE_UPDATE = "AdaptiveUpdate"
    END = "End"

File: tests/functional/Bots/DotNet/Skills/CodeFirst/WaterfallSkillBot/Dialogs/Cards/CardOptions.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace Microsoft.BotFrameworkFunctionalTests.WaterfallSkillBot.Dialogs.Cards
{
    public enum CardOptions
    {
        /// <summary>
        /// Adaptive card - Bot action
        /// </summary>
        AdaptiveCardBotAction,

        /// <summary>
        /// Adaptive card - Task module
        /// </summary>
        AdaptiveCardTeamsTaskModule,

        /// <summary>
        /// Adaptive card - Submit action
        /// </summary>
        AdaptiveCardSubmitAction,

        /// <summary>
        /// Hero cards
        /// </summary>
        Hero,

        /// <summary>
        /// Thumbnail cards
        /// </summary>
        Thumbnail,

        /// <summary>
        /// Receipt cards
        /// </summary>
        Receipt,

        /// <summary>
        /// Signin cards
        /// </summary>
        Signin,

        /// <summary>
        /// Carousel cards
        /// </summary>
        Carousel,

        /// <summary>
        /// List cards
        /// </summary>
        List,

        /// <summary>
        /// O365 cards
        /// </summary>
        O365,

        /// <summary>
        /// File cards
        /// </summary>
        TeamsFileConsent,

        /// <summary>
        /// Animation cards
        /// </summary>
        Animation,

        /// <summary>
        /// Audio cards
        /// </summary>
        Audio,

        /// <summary>
        /// Video cards
        /// </summary>
        Video,

        /// <summary>
        /// Adaptive update cards
        /// </summary>
        AdaptiveUpdate,

        /// <summary>
        /// Ends the card selection dialog
        /// </summary>
        End
    }
}
